src/noisepy/seis/scedc_s3store.py

```diff
@@ -72,12 +74,13 @@ def read_data(self, timespan: DateTimeRange, chan: Channel) -> ChannelData:
             f"{chan.station.network}{chan.station.name.ljust(5, '_')}{chan.type.name}"
             f"{chan.station.location.ljust(3, '_')}"
         )
-        filename = os.path.join(self.directory, f"{chan_str}{timespan.start_datetime.strftime('%Y%j')}.ms")
-        if not os.path.exists(filename):
+        filename = os.path.join(self.path, f"{chan_str}{timespan.start_datetime.strftime('%Y%j')}.ms")
```

Comment:
This problem is such a mess. I've had to deal with this a lot for echopype development. A utility module that sanitizes or checks for this would be best. As an example, we have done this (though we haven't quite cleaned this up 😬 ): https://github.com/OSOceanAcoustics/echopype/blob/43a66bb4fca880e4ae52176c14b6a5f1781bcab1/echopype/utils/io.py#L236-L240. Additionally for anything with URL, you can do a check for `://` in the string since all url is built that way, and force the path join to be with `/` by doing the python string join method.